backend/routers/slack.router.js

```diff
@@ -24,85 +20,53 @@ if (process.env.NODE_ENV !== "test") {
 }
 
 //Finds Id number of channel
-router.get("/findId", (req, res) => {
-  publishMessage();
-  findEvent();
-  // findProject();
-});
+router.get("/scheduleMessage", findCurrentEvents, scheduleMessage);
 
-//uses Id number to send message to said channel
-router.post("/postMeeting/:id", (req, res) => {
-  publishMessage1();
-});
+async function scheduleMessage(req, res, next) {
+  let channel;
+  let startTime;
+  let zoomChannel;
+  let events = res.locals.events;
+  events.forEach((cur) => {
+    let channelUrlArray = cur.project.slackUrl.split("/");
+    channel = channelUrlArray[channelUrlArray.length - 1];
+    startTime = cur.startTime;
+    zoomChannel = cur.project.videoConferenceLink;
+  });
+  startTime = startTime / 1000 - 3600000;
+  let currentTime = Math.round(new Date().getTime() / 1000);
 
-async function findConversation(name) {
   try {
-    const result = await app.client.conversations.list({
-      token: process.env.SLACK_BOT_TOKEN,
+    const result = await app.client.chat.scheduleMessage({
+      // The token you used to initalize the app
+      token: process.env.SLACK_OAUTH_TOKEN,
+      channel: channel,
+      text: "Slack Message Publish SCHEDULED",
+      // Time to post message, in Unix Epoch timestamp format
+      post_at: startTime,
     });
 
-    for (var channel of result.channels) {
-      if (channel.name === name) {
-        conversationId = channel.id;
-
-        console.log("Found conversation ID: " + conversationId);
-        break;
-      }
-    }
+    // Print result
   } catch (error) {
     console.error(error);
   }
 }
 
-async function publishMessage(id, text) {
-  try {
-    const result = await app.client.chat.postMessage({
-      token: process.env.SLACK_BOT_TOKEN,
-      channel: process.env.SLACK_CHANNEL_ID,
-      text: "Slack Message Publish",
-    });
-
-    console.log(result);
-  } catch (error) {
-    console.error(error);
-  }
-}
-
-async function publishMessage1(id, text) {
-  try {
-    const result = await app.client.chat.postMessage({
-      token: process.env.SLACK_BOT_TOKEN,
-      channel: id,
-      text: text,
-    });
-
-    console.log(result);
-  } catch (error) {
-    console.error(error);
-  }
-}
-
-async function findEvent(req, res) {
-  Event.find({})
+function findCurrentEvents(req, res, next) {
+  Event.find()
```

Comment:
I am not sure on what the performance of this query is, but we can do this in one call versus multiple. We can send a query to the DB to only return values where the date is greater than the date you want. The DB will then return a smaller query. We would not need to iterate over the longer array.

Reply:
@nickbeaird There are no multiple calls here. There is still the one call but it queries all events from the DB.
@Jonathanko52 In other words please add a condition for a date into the `find` method.

Reply:
@alex-anakin, this function has been rewritten from an enpoint (req, res) and made into middleware (req, res, next). That makes this call now multiple calls, as it will be passed onto the next call. https://expressjs.com/en/guide/using-middleware.html.